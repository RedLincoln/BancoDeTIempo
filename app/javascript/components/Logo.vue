<template>
  <v-img :max-width="maxWidth" :src="src"> </v-img>
</template>

<script>
import logoDark from "images/logo_dark.png";
import logo from "images/logo.png";

export default {
  props: {
    maxWidth: {
      type: String,
      default: "150",
    },
    dark: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    src() {
      return this.dark ? logoDark : logo;
    },
  },
};
</script>

<style>
</style>