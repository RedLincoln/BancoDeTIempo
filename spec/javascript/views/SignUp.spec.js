import Vuetify from "vuetify";
import Vuex from "vuex";

import SignUp from "../../../app/javascript/views/SingUp.vue";
import Logo from "../../../app/javascript/components/Logo.vue";
import { createLocalVue, mount } from "@vue/test-utils";

describe("SignUp.vue", () => {
  const localVue = createLocalVue();
  localVue.component("Logo", Logo);
  let vuetify;
  let wrapper;

  beforeEach(() => {
    localVue.use(Vuex);

    vuetify = new Vuetify();
    wrapper = mount(SignUp, {
      localVue,
      vuetify,
    });
  });

  it("renders properly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("have all the necesary fields", () => {
    expect(wrapper.find('[data-testid="name-field"]').exists()).toBeTruthy();
    expect(
      wrapper.find('[data-testid="information-field"]').exists()
    ).toBeTruthy();
    expect(wrapper.find('[data-testid="email-field"]').exists()).toBeTruthy();
    expect(wrapper.find('[data-testid="avatar-field"]').exists()).toBeTruthy();
  });
});
