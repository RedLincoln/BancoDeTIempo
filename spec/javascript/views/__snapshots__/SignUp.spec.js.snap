// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp.vue renders properly 1`] = `
<div>
  <form novalidate="novalidate" class="v-form">
    <div class="v-input theme--light v-text-field">
      <div class="v-input__control">
        <div class="v-input__slot">
          <div class="v-text-field__slot"><input data-testid="name-field" id="input-3" type="text"></div>
        </div>
        <div class="v-text-field__details">
          <div class="v-messages theme--light">
            <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
          </div>
        </div>
      </div>
    </div>
    <div class="v-input theme--light v-text-field">
      <div class="v-input__control">
        <div class="v-input__slot">
          <div class="v-text-field__slot"><input data-testid="email-field" id="input-6" type="text"></div>
        </div>
        <div class="v-text-field__details">
          <div class="v-messages theme--light">
            <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
          </div>
        </div>
      </div>
    </div>
    <div class="v-input theme--light v-text-field v-file-input">
      <div class="v-input__prepend-outer">
        <div class="v-input__icon v-input__icon--prepend"><button type="button" aria-label="prepend icon" class="v-icon notranslate v-icon--link mdi mdi-paperclip theme--light"></button></div>
      </div>
      <div class="v-input__control">
        <div class="v-input__slot">
          <div class="v-text-field__slot">
            <div class="v-file-input__text"></div><input data-testid="avatar-field" id="input-9" type="file">
          </div>
          <div class="v-input__append-inner">
            <div class="v-input__icon v-input__icon--clear"><button disabled="disabled" type="button" aria-label="clear icon" class="v-icon notranslate v-icon--disabled v-icon--link mdi mdi-close theme--light"></button></div>
          </div>
        </div>
        <div class="v-text-field__details">
          <div class="v-messages theme--light">
            <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
          </div>
        </div>
      </div>
    </div>
    <div class="v-input v-textarea theme--light v-text-field">
      <div class="v-input__control">
        <div class="v-input__slot">
          <div class="v-text-field__slot"><textarea data-testid="information-field" id="input-14" rows="5"></textarea></div>
        </div>
        <div class="v-text-field__details">
          <div class="v-messages theme--light">
            <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
